ministryofjustice/analytics-platform-control-panel
# ANPL-1049-apps-create-disable-authentication-functionality-in-control-panel

## :memo: Summary
This PR allows the admin user to add additional functionality to the Create App page.
The field disable_authentication is used to switch off or on authentication for this app.
The above key is used in the deployment process of said app.

Traditionally, this field is set by devops however this feature allows us to delegate this job to app admins to later be used in the deployment process.

This PR ...
- Adds disable_authentication key to AWS Secrets Manager
- Frontend gui added to create app page
- Django form field added with value to pop and used in just django Form

The changes in this PR are needed because ...
it will move the field disable_authentication to AWS secrets manager instead of hard coded in deployment

Merging this PR will have the following side-effects:
Admin users can now use the GUI on create app page to add this parameter field instead of relying on Devops.

## :mag: What should the reviewer concentrate on?
- very basic, field added to form
- piggybacked addition on to existing addition, namely 'client secret data'
- double check that the help text makes sense and nothing more detailed is required

## :technologist: How should the reviewer test these changes?
- open create app as admin
- tick disable_authentication checkbox
- go to secret manager on AWS and check that this field has been added
- if unchecked then there shouldn't be anything on AWS secrets manager

## :books: Documentation status
